### ISSUE TITLE
--no-update-deps is not respected

### ISSUE THREAD

**Issue by jakirkham** [conda/conda#7959]
## Current Behavior

Running `conda install -c conda-forge --no-update-deps --quiet --yes vs2008_express_vc_python_patch` in a new copy of Miniconda with some packages updated from `defaults` results in downgrading a bunch of packages and switching channels.

### Steps to Reproduce

```
conda install -c conda-forge --no-update-deps --quiet --yes vs2008_express_vc_python_patch
```

ref: https://ci.appveyor.com/project/jakirkham/cybuffer/builds/20303538/job/nq64t0ak2yyp16sa#L179

## Expected Behavior

Only `vs2008_express_vc_python_patch` should be installed. Nothing else should be changed.

## Environment Information

Unfortunately part of the issue is that `conda` has messed up the `base` environment here. So this is about as good as I can do. Unfortunately this is also showing up after the change. Though it looks like `conda` itself was spared.

<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : C:\Miniconda36
            shell level : 1
       user config file : C:\Users\appveyor\.condarc
 populated config files : C:\Users\appveyor\.condarc
          conda version : 4.5.11
    conda-build version : 3.16.2
         python version : 3.6.5.final.0
       base environment : C:\Miniconda36  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-32
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-32
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-32
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/win-32
                          https://repo.anaconda.com/pkgs/pro/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-32
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Miniconda36\pkgs
                          C:\Users\appveyor\AppData\Local\conda\conda\pkgs
       envs directories : C:\Miniconda36\envs
                          C:\Users\appveyor\AppData\Local\conda\conda\envs
                          C:\Users\appveyor\.conda\envs
               platform : win-32
             user-agent : conda/4.5.11 requests/2.20.0 CPython/3.6.5 Windows/2012ServerR2 Windows/6.3.9600
          administrator : True
             netrc file : None
           offline mode : False
```
</p></details>

**Comment by jakirkham** [comment]
Switching to `conda install --no-update-deps --quiet --yes conda-forge::vs2008_express_vc_python_patch` improves the situation, but still seems to be touching packages that should be left alone.

ref: https://ci.appveyor.com/project/jakirkham/cybuffer/builds/20304187/job/nxthuriovq7g8wac#L153

**Comment by astrojuanlu** [comment]
This also happens when trying to do

```
conda install --no-update-deps pip
```

The `--no-update-deps` flag is completely ignored:

```
(base) root@6030642ddcfa:/# conda install --no-update-deps pip
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /root/miniconda3

  added / updated specs:
    - pip


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2021.7.5   |       h06a4308_1         113 KB
    certifi-2021.5.30          |   py39h06a4308_0         139 KB
    conda-4.10.3               |   py39h06a4308_0         2.9 MB
    openssl-1.1.1l             |       h7f8727e_0         2.5 MB
    pip-21.2.4                 |   py37h06a4308_0         1.8 MB
    ------------------------------------------------------------
                                           Total:         7.4 MB

The following packages will be UPDATED:

  ca-certificates                      2020.12.8-h06a4308_0 --> 2021.7.5-h06a4308_1
  certifi                          2020.12.5-py39h06a4308_0 --> 2021.5.30-py39h06a4308_0
  conda                                4.9.2-py39h06a4308_0 --> 4.10.3-py39h06a4308_0
  openssl                                 1.1.1i-h27cfd23_0 --> 1.1.1l-h7f8727e_0
  pip                                 20.3.1-py39h06a4308_0 --> 21.2.4-py37h06a4308_0


Proceed ([y]/n)? n

```

as reported in https://github.com/pyenv/pyenv/issues/2070. I tried it with conda 4.7 and 4.9. At the moment it's not possible to know whether 4.10 exhibits the same behaviour, because in https://repo.anaconda.com/miniconda/ there are only installers for 4.10.3 for x86_64, and therefore the upgrade doesn't happen because there is no newer version.

**Comment by jakirkham** [comment]
Seem to recall this coming up in the RTD context as well, but not recalling the relevant issue # atm.

**Comment by astrojuanlu** [comment]
To clarify, this affects all conda users (as far as I understand), even outside Read the Docs. But probably @humitos will know if we've dealt with this in the past already.

**Comment by humitos** [comment]
@jakirkham I think you refer to this issue https://github.com/readthedocs/readthedocs.org/issues/3829 (that you are the author 😄 )

**Comment by astrojuanlu** [comment]
Is `conda config --set auto_update_conda False` the official workaround for this issue?

**Comment by astrojuanlu** [comment]
Sorry, dropped the comment too early. Let me rephrase:

I'm not sure if `--no-update-deps` is being ignored in all cases nowadays, or if the conda auto-update is the only situation that overrides it. Therefore, I'm wondering if `conda config --set auto_update_conda False` is either (a) a possible workaround, or (b) _the_ official solution.

**Comment by jakirkham** [comment]
> @jakirkham I think you refer to this issue [readthedocs/readthedocs.org#3829](https://github.com/readthedocs/readthedocs.org/issues/3829) (that you are the author 😄 )

Oh man, wow! 😂 Thank you for reminding me. Sorry for forgetting 😅

**Comment by astrojuanlu** [comment]
Still present:

```
(base) root@f9f7e8f76947:/# conda --version
conda 4.12.0
(base) root@f9f7e8f76947:/# conda install --no-update-deps pip
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /root/miniconda3

  added / updated specs:
    - pip


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2022.07.19 |       h06a4308_0         124 KB
    certifi-2022.9.14          |   py39h06a4308_0         155 KB
    conda-4.14.0               |   py39h06a4308_0         915 KB
    cytoolz-0.11.0             |   py39h27cfd23_0         350 KB
    openssl-1.1.1q             |       h7f8727e_0         2.5 MB
    pip-22.1.2                 |   py39h06a4308_0         2.5 MB
    toolz-0.11.2               |     pyhd3eb1b0_0          49 KB
    ------------------------------------------------------------
                                           Total:         6.5 MB

The following NEW packages will be INSTALLED:

  cytoolz            pkgs/main/linux-64::cytoolz-0.11.0-py39h27cfd23_0
  toolz              pkgs/main/noarch::toolz-0.11.2-pyhd3eb1b0_0

The following packages will be UPDATED:

  ca-certificates                      2022.3.29-h06a4308_1 --> 2022.07.19-h06a4308_0
  certifi                          2021.10.8-py39h06a4308_2 --> 2022.9.14-py39h06a4308_0
  conda                               4.12.0-py39h06a4308_0 --> 4.14.0-py39h06a4308_0
  openssl                                 1.1.1n-h7f8727e_0 --> 1.1.1q-h7f8727e_0
  pip                                 21.2.4-py39h06a4308_0 --> 22.1.2-py39h06a4308_0


Proceed ([y]/n)? 
```

**Comment by astrojuanlu** [comment]
I've tried to reproduce this issue. Half of the times I couldn't (which is good), the other half I got a soup of symbols that I couldn't understand:

```
UnsatisfiableError: The following specifications were found
to be incompatible with the existing python installation in your environment:

Specifications:

  - conda[version='<4.6'] -> python[version='>=2.7,<2.8.0a0|>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.5,<3.6.0a0|>=3.4,<3.5.0a0']

Your python: python=3.9
...

The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.35=0
  - feature:|@/linux-64::__glibc==2.35=0

```

(This is from a freshly installed Miniconda)

In any case, I cannot prove that the bug is still present.
